pkg/scheduler/core/generic_scheduler.go

```diff
@@ -281,50 +278,64 @@ func (g *genericScheduler) Preempt(ctx context.Context, prof *profile.Profile, s
 		klog.Infof("%v potential nodes for preemption, first %v are: %v", len(potentialNodes), len(sample), sample)
 	}
 	var pdbs []*policy.PodDisruptionBudget
-	if g.pdbLister != nil {
-		pdbs, err = g.pdbLister.List(labels.Everything())
+	pdbLister := fh.SharedInformerFactory().Policy().V1beta1().PodDisruptionBudgets().Lister()
+	if pdbLister != nil {
+		pdbs, err = pdbLister.List(labels.Everything())
 		if err != nil {
-			return "", nil, nil, err
+			return "", err
 		}
 	}
-	nodeNameToVictims, err := selectNodesForPreemption(ctx, prof, g.podNominator, state, pod, potentialNodes, pdbs)
+	nodeNameToVictims, err := selectNodesForPreemption(ctx, fh.PreemptHandle(), fh.PreemptHandle(), state, pod, potentialNodes, pdbs)
```

Comment:
Could we make `selectNodesForPreemption` as public method so that user's custom preemption plugins can reuse this method?

Reply:
It will be made public in a followup PR. This PR just keep the old interfaces as is.

Reply:
Got it, thanks!